src/ui/src/mainwindow.cpp

```diff
@@ -18,27 +25,74 @@ MainWindow::MainWindow(QWidget* parent)
 	shape_list_widget_->setWindowTitle("Shapes");
 	// metadata widget
 	metadata_widget_ = new MetadataWidget(this);
```

Comment:
1. Consider move member initialization in initializer list
2. This ctor is already pretty big and does a bunch of different things, maybe split it somehow?